packages/xo-web/src/common/render-xo-item.js

```diff
@@ -17,6 +20,104 @@ const OBJECT_TYPE_TO_ICON = {
   network: 'network',
 }
 
+const XoItem = ({ children, item, link, redirect, to }) =>
+  item !== undefined ? (
+    link ? (
+      <Link to={to} target={redirect ? undefined : '_blank'}>
+        {children()}
+      </Link>
+    ) : (
+      children()
+    )
+  ) : (
+    <span className='text-muted'>{_('errorNoSuchItem')}</span>
+  )
+
+// ===================================================================
+
+const XO_ITEM_PROP_TYPES = {
+  id: PropTypes.string.isRequired,
+  link: PropTypes.bool,
+  redirect: PropTypes.bool,
+}
+
+export const VmItem = [
+  connectStore(() => {
+    const getVm = createGetObject((_, { id }) => id)
+    return {
+      vm: getVm,
+      container: createGetObject(
+        createSelector(getVm, vm => get(() => vm.$container))
+      ),
+    }
+  }),
+  ({ vm, container, ...props }) => (
+    <XoItem item={vm} to={`/vms/${vm.id}`} {...props}>
+      {() => (
+        <span>
+          <Icon icon={`vm-${vm.power_state.toLowerCase()}`} />{' '}
+          {vm.name_label || vm.id}
+          {container !== undefined &&
+            ` (${container.name_label || container.id})`}
+        </span>
+      )}
+    </XoItem>
+  ),
+].reduceRight((value, decorator) => decorator(value))
+
+VmItem.propTypes = XO_ITEM_PROP_TYPES
```

Comment:
Why don't you assign those to `XoItem` directly and add proptypes specific to `VmItem` here (there shouldn't be many).

Reply:
They have the same propTypes. To avoid duplication, i mutualise the propTypes

Reply:
Ok, but if they all have the same propTypes, why not using `XoItem.propTypes` and not using any of the `*Item`'s propTypes? The warning would correctly show in the console.